### ISSUE TITLE
[Enhancement]: Play count

### ISSUE THREAD

**Issue by burnshroom** [advplyr/audiobookshelf#1653]
### Describe the feature/enhancement

I mentioned this last year under [issue #527](https://github.com/advplyr/audiobookshelf/issues/527#issuecomment-1152207707).

Still holding out hope for it as I have several books I listen to over and over again and would like to see how many times I may have accomplished this.  

Similar to what Plex has and that it also allows you to sort by this play count.  

![image](https://user-images.githubusercontent.com/123109/230566966-f988bfa5-4004-4bc5-bb14-51af08a82465.png)

Not something that makes or breaks the app at all, but would be nice to have.

**Comment by advplyr** [comment]
The difficulty I have with this is how is a play counted? What if you toggle the "Mark as Finished" button a few times, would those be plays?
Maybe testing what Plex considers a play would be helpful

**Comment by Dr-Blank** [comment]
Here is how plex counts the plays:

```mermaid
flowchart TD
    A["Play count = 0 (Unwatched State)"] -->|Watch more than 90%| B["Play count = 1"]
    A["Play count = 0 (Unwatched State)"] -->|Mark it as watched| B["Play count = 1"]
    B -->|Watch more than 90%| C["Play count +=1"]
    C -->|Watch more than 90%| C["Play count +=1"]
    C -->|Mark as unwatched| A
```

**Comment by Jmanko16** [comment]
Honestly a simple counter (with a plus/minus) would work.  So if you mark watched etc but did on accident jus hit the negative button.  Obviously have the play count update on finished as above as well but this would give more control (and allow for reading outside ABS If used as a front end for audiobook/ebook)

**Comment by nichwall** [comment]
To track the plays, I like the idea of the plus/minus buttons to simplify tracking and accidentally "Making as Finished" messing up the counter.

Incorporating the idea from #1905, could this be added as an array of completion times (separate item entry in the database)? My idea for the UI would look something like the History view in the Android app (for checking sync status), where each line is the "completion date", and then have an Edit and Remove button on the right to fix errors easily. In a web browser, it would just be a popup because that information doesn't need a ton of horizontal space.

To get to this view, the user can either use the three dot menu OR click on the listening progress box.

![Screenshot_20230715-073638.png](https://github.com/advplyr/audiobookshelf/assets/5686638/3a2757b5-f9c9-428a-a61e-6018e10a182e)

This array would be tracked separately from Listen Progress/Finished, so if I had read the book before and wanted to reread it, it would no longer be marked as finished (and show up on the Continue Listening Shelf), but still show up as "Read X time(s)" by counting how many elements are in the array.

I'll try and make a mock-up of the UI later today, not sure if I'll have time though

I'm still thinking through how/whether this should differentiate ebook and audiobook progress

**Comment by Cpnkrk** [comment]
I agree with adding a counter but do not get rid of the option of manually marking as completed due to maybe listening to a book on another app or something.
